docs/tutorials/nextjs.mdx

```diff
@@ -0,0 +1,186 @@
+
+RSCs abilitiy to block for data means that with the App Router you no longer have `getServerSideProps` to fetch data for rendering. Any component in the tree can make asychronous requests for data. While this is very convenient it also means thats if you define global variables (like the Redux store) they will be shared across requests. This is a problem because the Redux store could be contaminated with data from other requests.
+
+### Creating a Redux Store per Request
+
+Following along with the (Quick-Start guide)[./quick-start.mdx] we need to make some changes to the `app/store.js` file. The first change is to move from defining store as a global to defining a `createStore` function that returns a new store for each request.
+
+```ts title="src/app/store.ts"
+import { configureStore } from '@reduxjs/toolkit'
+
+export const createStore = () =>
+  configureStore({
+    reducer: {},
+  })
+
+// Infer the type of createStore
+export type StoreType = ReturnType<typeof createStore>
+// Infer the `RootState` and `AppDispatch` types from the store itself
+export type RootState = ReturnType<StoreType['getState']>
+// Inferred type: {posts: PostsState, comments: CommentsState, users: UsersState}
+export type AppDispatch = StoreType['dispatch']
+```
+
+To use this new `createStore` function we need to create a new "client" component that will create the store and share it using the React-Redux `Provider` component.
+
+```ts title="src/app/StoreProvider.tsx"
+// file: app/store.ts noEmit
+import { configureStore } from '@reduxjs/toolkit'
+
+// highlight-start
+export const createStore = () =>
+  configureStore({
+    reducer: {},
+  })
+// highlight-end
+
+// Infer the type of createStore
+// highlight-start
+export type StoreType = ReturnType<typeof createStore>
+// highlight-end
+// Infer the `RootState` and `AppDispatch` types from the store itself
+export type RootState = ReturnType<StoreType['getState']>
+// Inferred type: {posts: PostsState, comments: CommentsState, users: UsersState}
+export type AppDispatch = StoreType['dispatch']
+
+// file: app/StoreProvider.tsx
+;('use client')
+import { useRef } from 'react'
+import { Provider } from 'react-redux'
+// highlight-start
+import { createStore } from './store'
+// highlight-end
+
+export default function StoreProvider({
+  children,
+}: {
+  children: React.ReactNode
+}) {
+  // highlight-start
+  const storeRef = useRef<ReturnType<typeof createStore>>(createStore())
+  // highlight-end
+
+  return <Provider store={storeRef.current}>{children}</Provider>
+}
+```
+
+If you need to intialize the store with data from the parent component then define that data as a property and use an action on the slice to set the data in the store as shown below.
```

Comment:
my concern with this approach is that it essentially endorses setter actions, which we explicitly discourage

my first approach would be to allow makeStore to receive a preloadedState option - dispatching actions like this would be okay as an alternative but i'd prefer something a little less "setter"-y for the example if possible

Reply:
I'd prefer the action, as it could open the door for an interesting data prefetching pattern - but I agree, it should be less `setter`-y.

Reply:
Yeah, I've done the "partial preloaded state" bit too in the one Next app I worked on, but I don't have access to that code any more. I'm sure I could find an example if I looked hard enough.